kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -181,13 +184,32 @@ impl<'tcx> GotocCtx<'tcx> {
                 let t = self.codegen_ty(pt);
                 let a = fargs.remove(0);
                 let b = fargs.remove(0);
+                let op_type = a.typ().clone();
                 let res = a.$f(b);
+                // add to symbol table
+                let res_type = arithmetic_overflow_result_type(op_type.clone());
+                let struct_tag =
+                    self.ensure_struct(res_type.tag().unwrap(), res_type.tag().unwrap(), |_, _| {
```

Comment:
Why do this here? I'd expect it to happen in `arithmetic_overflow_result_type`

Reply:
I was hoping to do it in `arithmetic_overflow_result_type`, but it doesn't have access to the Gotoc context since it's in the `cprover_bindings` crate.

Reply:
maybe make a helper function

```rust
let res_type = self.codegen_arithmetic_overflow_result_type(op_type.clone()); // does the ensure
let struct_tag = res_type.tag().unwrap()
```

Reply:
Makes sense. Done.